Realm.Shared/Dynamic/MetaRealmObject.cs

```diff
@@ -0,0 +1,225 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace Realms.Dynamic
+{
+    internal class MetaRealmObject : DynamicMetaObject
+    {
+        private readonly Realm _realm;
+        private readonly RealmObject.Metadata _metadata;
+
+        private static readonly FieldInfo RealmObjectRealmField = typeof(RealmObject).GetTypeInfo().GetField("_realm", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
+        private static readonly FieldInfo RealmObjectRowHandleField = typeof(RealmObject).GetTypeInfo().GetField("_rowHandle", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
+
+        public MetaRealmObject(Expression expression, DynamicRealmObject value)
+            : base(expression, BindingRestrictions.Empty, value)
+        {
+            _realm = value.Realm;
+            _metadata = value.ObjectMetadata;
+        }
+
+        public override DynamicMetaObject BindGetMember(GetMemberBinder binder)
+        {
+            var property = _metadata.Schema.Find(binder.Name);
+            if (property == null)
+            {
+                return base.BindGetMember(binder);
+            }
+
+            var convertedExpression = Expression;
+            if (convertedExpression.Type != LimitType)
+            {
+                convertedExpression = Expression.Convert(convertedExpression, LimitType);
+            }
+
+            var arguments = new List<Expression>
+            {
+                WeakConstant(_metadata.Table),
+                Expression.Field(convertedExpression, RealmObjectRowHandleField),
+                Expression.Constant(_metadata.ColumnIndices[property.Name])
+            };
+
+            MethodInfo realmObjectOpsMethod = null;
```

Comment:
I think you can avoid some code duplication by inferring just the method name rather than the whole info (as all of them seem to get the info from `typeof(RealmObjectOps).GetMethod({name})`